### PR TITLE
Adjust spine basic API

### PR DESCRIPTION
**Pull request by johanzhu**
What does this PR do?
Adjust spine basic API.

Changes
- Deprecated resource getter and setter
- Refractor SkeletonResource. Change name to spineResource, add an another animationData create from previous skeletonData.
Additionally create and store animationData, which is generated based on the original skeletonData. Add a skeletonData cache to manage resource reference counting relationships based on skeletonData.
- Add createSpineResource API and export it.
- Delete addSeparateSlot API
- Modify the renderer initialization method to initialize directly by setting Skeleton and State.
- Add static methods createWithEntity and attachToEntity to facilitate manual creation of Spine animations for users.
- Remove the setting variable and move the rendering parameters zSpacing and useClipping to the outermost level of the renderer. 

Impact：
After deprecating the resource API, the recommended way for users to create Spine animations will change."

old: 
```typescript
const spineResource = (await engine.resourceManager.load({
  ...resource,
  type: 'spine'
})) as SpineResource;
const spineEntity = new Entity(engine, 'spine-entity');
const spineAnimation = spineEntity.addComponent(SpineAnimationRenderer);
pineAnimation.resource = spineResource;
root.addChild(spineEntity);
```

new:
```typescript
const spineResource = (await engine.resourceManager.load({
  ...resource,
  type: 'spine'
})) as SpineResource;

// create with entity
const spineAnimation = SpineAnimationRenderer.createWithEntity(spineResource);
root.addChild(spineAnimation.entity);

// or  attach to entity
const spineEntity = scene.findEntityByName('spine');
const spineAnimation = SpineRendererAnimation.attachToEntity(spineEntity, spineResource);

// or  create with skeleton and state
// In this case, use should create new skeleton and AnimationState with source data in spineResource
const spineEntity = new Entity(engine, 'spine-entity');
const spineAnimation = spineEntity.addComponent(SpineAnimationRenderer);
const { skeletonData, animationData } = spineResource;
const skeleton = new Skeleton(skeletonData);
const state = new AnimationState(animationData);
spineAnimation.skeleton = skeleton;
spineAnimation.state = state;
root.addChild(spineEntity);

```

Additional Context
Related to issue: #109 

Please review the changes and provide feedback on the caching logic and implementation details.

### CODE REVIEW

File: src/loader/SpineResource.ts [hunk 1]
```diff
@@ -1,28 +1,42 @@
-import { SkeletonData } from "@esotericsoftware/spine-core";
+import { AnimationStateData, Skeleton, SkeletonData } from "@esotericsoftware/spine-core";
```
Comment (GuoLei1990) [inline src/loader/SpineResource.ts]: Not clear import

File: src/loader/SpineResource.ts [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
   private _skeletonData: SkeletonData;
+  private _animationData: AnimationStateData;
 
-  get skeletonData() {
+  /** @internal */
+  static addRefCount(skeletonData: SkeletonData, count: number) {
+    const resource = SpineResource._cache.get(skeletonData);
+    // @ts-ignore
+    resource._addReferCount(count);
+  }
```
Comment (GuoLei1990) [inline src/loader/SpineResource.ts:26]: Reference counting should not be specialized, use Templet design by `_markAsTemplate`

File: src/loader/SpineResource.ts [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
+
+  get skeletonData(): SkeletonData {
     return this._skeletonData;
   }
 
+  get animationData(): AnimationStateData {
```
Comment (GuoLei1990) [inline src/loader/SpineResource.ts]: Don't write comments?

File: src/loader/SpineResource.ts [hunk 4]
```diff
@@ -1,28 +1,42 @@
-import { SkeletonData } from "@esotericsoftware/spine-core";
+import { AnimationStateData, Skeleton, SkeletonData } from "@esotericsoftware/spine-core";
 import { Engine, ReferResource, Texture2D } from "@galacean/engine";
-import { SpineAnimationRenderer } from "../SpineAnimationRenderer";
 
-export class SkeletonDataResource extends ReferResource {
-  readonly textures: Texture2D[] = [];
+export class SpineResource extends ReferResource {
```
Comment (GuoLei1990) [inline src/loader/SpineResource.ts:9]: Class comments missed!

Comment (GuoLei1990) [inline src/loader/SpineResource.ts:9]: Try add instantiate method to improve useless!

File: src/loader/SpineResource.ts (hunk 1 above)
Comment (johanzhu) [reply src/loader/SpineResource.ts]: done.

File: src/loader/SpineResource.ts (hunk 2 above)
Comment (johanzhu) [reply src/loader/SpineResource.ts:26]: done. Already delete this function.

File: src/loader/SpineResource.ts (hunk 3 above)
Comment (johanzhu) [reply src/loader/SpineResource.ts]: done.

File: src/loader/SpineResource.ts (hunk 4 above)
Comment (johanzhu) [reply src/loader/SpineResource.ts:9]: done.

Comment (johanzhu) [reply src/loader/SpineResource.ts:9]: done.

File: src/loader/SpineResource.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+  /**
+   * Gets the skeleton data associated with this Spine resource.
+   * 
+   * This getter provides access to the `SkeletonData` used in constructing
+   * the skeleton entity template. It contains the static configuration for
+   * bones, slots, skins, events, and animations. External components or systems 
+   * can use this data to access and manipulate the structure and animations of the skeleton.
+   * 
+   * @returns {SkeletonData} The skeleton data for this resource.
```
Comment (GuoLei1990) [inline src/loader/SpineResource.ts]: Wrong comment specification

Comment (johanzhu) [reply src/loader/SpineResource.ts]: done.

File: src/loader/SpineResource.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+   * 
+  */
+  get skeletonData(): SkeletonData {
+    return this._skeletonData;
+  }
+
+  /**
+   * The animation state data associated with this Spine resource.
+  */
+  get animationData(): AnimationStateData {
```
Comment (GuoLei1990) [inline src/loader/SpineResource.ts]: `animationData`->`animationStateData`

File: src/loader/SpineResource.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        const attachment = Object.values(attachmentMap)[0];
+        // @ts-ignore
+        const texture = attachment?.region?.texture.texture;
+        if (texture && !this._texturesInSpineAtlas.find(item => item.instanceId === texture.instanceId)) {
+          this._texturesInSpineAtlas.push(texture);
+          texture._associationSuperResource(this);
+        }
+      });
+    });
+  }
```
Comment (GuoLei1990) [inline src/loader/SpineResource.ts:88]:
```ts
   private _associationTextureInSkeletonData(skeletonData: SkeletonData): void {
    const { skins, slots } = skeletonData;
    const textures = this._texturesInSpineAtlas;

    for (let i = 0, n = skins.length; i < n; i++) {
      for (let j = 0, m = slots.length; j < m; j++) {
        const slot = slots[j];
        const attachment = skins[i].getAttachment(slot.index, slot.name);

        const texture = <Texture2D>(<RegionAttachment>attachment)?.region.texture.texture;
        if (texture) {
          if (!textures.includes(texture)) {
            textures.push(texture);
            // @ts-ignore
            texture._associationSuperResource(this);
          }
        }
      }
    }
  }
```

Comment (johanzhu) [reply src/loader/SpineResource.ts:88]: done

File: src/loader/SpineResource.ts (hunk 6 above)
Comment (johanzhu) [reply src/loader/SpineResource.ts]: done

**Pull request merged**